File: generate-audio.py
import asyncio
import edge_tts
import os
import re

audio_dir = os.path.join(os.path.dirname(__file__), 'audio')
if not os.path.exists(audio_dir):
    os.makedirs(audio_dir)
    print('Created audio directory')

# Read cards-data.js
with open(os.path.join(os.path.dirname(__file__), 'cards-data.js'), 'r', encoding='utf-8') as f:
    content = f.read()

# Extract all cards
pattern = r"\{\s*letter:\s*'([^']+)',\s*word:\s*'([^']+)',[\s\S]*?audioWord:\s*'([^']+)',\s*audioSentence:\s*'([^']+)'"
matches = re.findall(pattern, content)

print(f'Found {len(matches)} cards\n')

async def generate_audio():
    for letter, word, audio_word, audio_sentence in matches:
        word_file = os.path.join(audio_dir, f'{word}_word.mp3')
        sentence_file = os.path.join(audio_dir, f'{word}_sentence.mp3')
        
        try:
            if audio_word:
                communicate = edge_tts.Communicate(audio_word, 'en-US-JennyNeural')
                await communicate.save(word_file)
                print(f'[OK] {word}_word.mp3')
            
            if audio_sentence:
                communicate = edge_tts.Communicate(audio_sentence, 'en-US-JennyNeural')
                await communicate.save(sentence_file)
                print(f'[OK] {word}_sentence.mp3')
        except Exception as e:
            print(f'[FAIL] {word}: {e}')
    
    print('\nAll audio files generated!')

asyncio.run(generate_audio())

File: generate-audio.js
const { toVoice, voices } = require("edge-tts-nodejs");
const fs = require('fs');
const path = require('path');

const audioDir = path.join(__dirname, 'audio');
if (!fs.existsSync(audioDir)) {
  fs.mkdirSync(audioDir);
  console.log('创建 audio 目录');
}

console.log('开始生成音频文件...\n');

const cardsDataFile = fs.readFileSync(path.join(__dirname, 'cards-data.js'), 'utf-8');

const cardsData = [];

const cardMatches = cardsDataFile.matchAll(/\{\s*letter:\s*['"]([^'"]+)['"],\s*word:\s*['"]([^'"]+)['"],[\s\S]*?audioWord:\s*['"]([^'"]+)['"],\s*audioSentence:\s*['"]([^'"]+)['"][\s\S]*?\}/g);

for (const match of cardMatches) {
  const letter = match[1];
  const word = match[2];
  const audioWord = match[3];
  const audioSentence = match[4];
  
  cardsData.push({ letter, word, audioWord, audioSentence });
}

console.log(`找到 ${cardsData.length} 个卡片\n`);

for (const card of cardsData) {
  const wordFile = path.join(audioDir, `${card.word}_word.mp3`);
  const sentenceFile = path.join(audioDir, `${card.word}_sentence.mp3`);
  
  try {
    if (card.audioWord) {
      toVoice(card.audioWord, wordFile, {
        voice: "en-US-JennyNeural"
      });
      console.log(`✓ ${card.word}_word.mp3`);
    }
    
    if (card.audioSentence) {
      toVoice(card.audioSentence, sentenceFile, {
        voice: "en-US-JennyNeural"
      });
      console.log(`✓ ${card.word}_sentence.mp3`);
    }
  } catch (error) {
    console.error(`✗ ${card.word} 生成失败:`, error.message);
  }
}

console.log('\n全部生成完成！');
